pkg/virt-controller/services/template.go

```diff
@@ -377,6 +430,18 @@ func (t *templateService) RenderLaunchManifest(vmi *v1.VirtualMachineInstance) (
 			},
 		})
 	}
+
+	if util.IsVhostuserVmi(vmi) {
+		if vmi.Spec.Domain.Memory == nil || vmi.Spec.Domain.Memory.Hugepages == nil {
+			logger := log.DefaultLogger()
+			err := fmt.Errorf("Hugepages is required for vhostuser interface")
```

Comment:
we'd want to catch things like this in our validation webhook during the REST call to post the object.

Where this check is occurring now, it happens after the VMI has been persisted. Ideally, we don't allow invalid specs to get persisted at all.

Reply:
Ok, then the correct place to move this check is to [vmi-create-admitter.go](https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go). Thanks, I will update the changes accordingly.